src/inmanta/ast/type.py

```diff
@@ -161,6 +200,43 @@ def __eq__(self, other: object) -> bool:
             return NotImplemented
         return self.element_type == other.element_type
 
+    def is_primitive(self) -> bool:
+        return self.element_type.is_primitive()
```

Comment:
This does not correspond with the docstring. As far as I can tell, this was an orphaned method before this PR, so I think we have some freedom, but I think we need to define the semantics of "primitive" properly in the docstring.

Reply:
updated